src/main/java/com/gzs/daos/inmemory/InMemoryDao.java

```diff
@@ -0,0 +1,40 @@
+package com.gzs.daos.inmemory;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class InMemoryDao<T> {
+
+    protected Map<Integer, T> dataMap;
+
+    public InMemoryDao() {
+        this.dataMap = new HashMap<Integer, T>();
+    }
+
+    protected List<T> getAllFromMap() {
+        List<T> data = new ArrayList<>(dataMap.values());
+        return data;
+    }
+
+    protected T getById(int id) {
+        T newT;
+        newT = dataMap.get(id);
+        return newT;
+    }
+
+    protected boolean insertT (T t, int id) {
```

Comment:
`id` is already contained in all your entity objects (Term, Translation...), so you could extract it from there. Problem is in the fact that this is a generic class and compiler doesn't know that it will always be some entity.

Since properties like `id` are shared by all of your entities, it would be convenient to be able to work with common properties somehow. I propose having a superclass which all your entity classes will extend (you can name it `DatabaseEntity`) or something similar. This class will have `id` (and other common fields/methods if applicable) within itself so you can work with all your entities in a generic way. Instead of `InMemory<T>`, you can put `InMemory<? extends DatabaseEntity>`.

Then you can write only one insert method which will receive DatabaseEntity and insert it into dataMap.

Reply:
@celicni Created DatabaseEntity.class and now insert/update/delete methods are defined in InMemoryDao.class.